File: app/src/main/java/simple/clever/notes/data/CardDataMapping.java
package simple.clever.notes.data;

import com.google.firebase.Timestamp;

import java.util.HashMap;
import java.util.Map;

public class CardDataMapping {

    public static class Fields{
        public final static String DATE = "date";
        public final static String HEAD = "title";
        public final static String FAVORITE = "favorite";
    }

    public static CardData toCardData(String id, Map<String, Object> doc) {
        Timestamp timeStamp = (Timestamp)doc.get(Fields.DATE);
        CardData answer = new CardData((String) doc.get(Fields.HEAD), timeStamp.toDate(), (boolean) doc.get(Fields.FAVORITE));
        answer.setId(id);
        return answer;
    }

    public static Map<String, Object> toDocument(CardData cardData){
        Map<String, Object> answer = new HashMap<>();
        answer.put(Fields.HEAD, cardData.getHead());
        answer.put(Fields.DATE, cardData.getTimeOpen());
        answer.put(Fields.FAVORITE, cardData.getFavorite());
        return answer;
    }
}

File: app/src/main/java/simple/clever/notes/data/CardDataMapping.kt
package simple.clever.notes.data

import com.google.firebase.Timestamp

import java.util.HashMap


class CardDataMapping {

    object Fields{
        const val DATE: String = "date"
        const val HEAD: String = "title"
        const val FAVORITE: String = "favorite"
    }


    companion object{
        @JvmStatic
         fun toCardData(id: String, doc: Map<String, Any>): CardData {
            val timeStamp: Timestamp = doc.get(Fields.DATE) as Timestamp
            val answer= CardData(doc.get(Fields.HEAD) as String, timeStamp.toDate(), doc.get(Fields.FAVORITE) as Boolean)
            answer.id = id
            return answer
        }
        @JvmStatic
        fun toDocument(cardData: CardData): MutableMap<String, Any>{
            val answer: MutableMap<String, Any> = HashMap()
            answer[Fields.HEAD] = cardData.head
            answer.put(Fields.DATE, cardData.timeOpen)
            answer.put(Fields.FAVORITE, cardData.favorite)
            return answer
        }
    }
}

